drivers/w1/w1_net.c

```diff
@@ -0,0 +1,405 @@
+	if (ret < 0) {
+		goto out;
+	}
+	ret = w1_read_block(dev, (uint8_t *)rom_id, sizeof(struct w1_rom));
+	if (ret < 0) {
+		goto out;
+	}
+	if (w1_crc8((uint8_t *)rom_id, sizeof(struct w1_rom)) != 0) {
+		ret = -EIO;
+	}
+
+out:
+	(void)w1_unlock_bus(dev);
+	return ret;
+};
+
+static int match_rom(const struct device *dev, const struct w1_config *config)
+{
+	int ret;
+	uint8_t cmd;
+
+	if (!config->overdrive) {
+		if (w1_configure(dev, W1_SETTING_SPEED, 0) < 0) {
+			return -EIO;
+		}
+	}
+
+	ret = w1_reset_bus(dev);
+	if (ret == 0) {
+		ret = -ENODEV;
+		goto out;
+	}
+	if (ret < 0) {
+		goto out;
+	}
+
+	cmd = config->overdrive ? W1_CMD_OVERDRIVE_MATCH_ROM : W1_CMD_MATCH_ROM;
+	ret = w1_write_byte(dev, cmd);
+	if (ret < 0) {
+		goto out;
+	}
+	ret = w1_write_block(dev, (uint8_t *)&config->rom_id, 8);
+	if (ret < 0) {
+		goto out;
+	}
+
+	if (config->overdrive) {
+		if (w1_configure(dev, W1_SETTING_SPEED, 1) < 0) {
+			return -EIO;
+		}
+	}
+out:
+	return ret;
+};
+
+int w1_match_rom(const struct device *dev, const struct w1_config *config)
+{
+	int ret;
+
+	(void)w1_lock_bus(dev);
+	ret = match_rom(dev, config);
+	(void)w1_unlock_bus(dev);
+	return ret;
+}
+
+int w1_resume_command(const struct device *dev)
+{
+	int ret;
+
+	(void)w1_lock_bus(dev);
+	ret = w1_reset_bus(dev);
+	if (ret == 0) {
+		ret = -ENODEV;
+		goto out;
+	}
+	if (ret < 0) {
+		goto out;
+	}
+
+	ret = w1_write_byte(dev, W1_CMD_RESUME);
+out:
+	(void)w1_unlock_bus(dev);
+	return ret;
+}
+
+static int skip_rom(const struct device *dev, const struct w1_config *config)
+{
+	int ret;
+	uint8_t cmd;
+
+	if (!config->overdrive) {
+		if (w1_configure(dev, W1_SETTING_SPEED, 0) < 0) {
+			return -EIO;
+		}
+	}
+
+	ret = w1_reset_bus(dev);
+	if (ret == 0) {
+		ret = -ENODEV;
+		goto out;
+	}
+	if (ret < 0) {
+		goto out;
+	}
+
+	cmd = config->overdrive ? W1_CMD_OVERDRIVE_SKIP_ROM : W1_CMD_SKIP_ROM;
+	ret = w1_write_byte(dev, cmd);
+	if (ret < 0) {
+		goto out;
+	}
+
+	if (config->overdrive) {
+		if (w1_configure(dev, W1_SETTING_SPEED, 0) < 0) {
+			return -EIO;
+		}
+	}
+
+out:
+	return ret;
+}
+
+int w1_skip_rom(const struct device *dev, const struct w1_config *config)
+{
+	int ret;
+
+	(void)w1_lock_bus(dev);
+	ret = skip_rom(dev, config);
+	(void)w1_unlock_bus(dev);
+	return ret;
+}
+
+static int reset_select(const struct device *dev, const struct w1_config *config)
+{
+	if (w1_get_peripheral_count(dev) > 1) {
+		return match_rom(dev, config);
+	}
+
+	return skip_rom(dev, config);
+}
+
+int w1_reset_select(const struct device *dev, const struct w1_config *config)
+{
+	int ret;
+
+	(void)w1_lock_bus(dev);
+	ret = reset_select(dev, config);
+	(void)w1_unlock_bus(dev);
+	return ret;
+}
+
+static int write_read(const struct device *dev, const struct w1_config *config,
+		      const uint8_t *write_buf, size_t write_len,
+		      uint8_t *read_buf, size_t read_len)
+{
+	int ret;
+
+	ret = reset_select(dev, config);
+	if (ret != 0) {
+		goto out;
```

Comment:
There are no locks in this function, no point in doing the `goto out`, will cost you as much as just `return` from here.